src/librustc_parse/parser/item.rs

```diff
@@ -111,13 +124,33 @@ impl<'a> Parser<'a> {
             .emit();
     }
 
+    /// Error in-case a `default` was parsed but no item followed.
+    fn error_on_unmatched_defaultness(&self, def: Defaultness) {
+        if let Defaultness::Default(span) = def {
+            self.struct_span_err(span, "unmatched `default`")
```

Comment:
I have the same issues with this wording as with "unmatched visibility", it's entirely unclear what it means.

Reply:
Maybe:
```rust
            self.struct_span_err(span, "`default` must be followed by an `fn`, `const`, or `type` item")
                .span_label(span, "the `default` qualifier")
                .emit();
```